.gitignore

```diff
@@ -1,4 +1,3 @@
 *
 
 !.gitignore
-!shell.c
```

Comment:
why you remove this line?

Reply:
Because I wanted to add the shell_substitution.h file when executing "git add" and it did not show. So I checked and found out that you tried to ignore almost everything except for shell.c. Thus I manually needed to add the header file in.

Reply:
I think the correct way should be to add !shell_substitution.h. Kudos for creating your functionality in separate file! I haven't thought about it implementing my part.